students/mkdir01/lesson07/test_html_render.py

```diff
@@ -0,0 +1,550 @@
+#!/usr/bin/env python
+
+from io import StringIO
+import html_render as hr
+
+
+def render_page(page, filename, indent=None):
+    f = StringIO()
+    if indent is None:
+        page.render(f)
+    else:
+        page.render(f, indent)
+
+    print(f.getvalue())
+    with open(filename, 'w') as outfile:
+        outfile.write(f.getvalue())
```

Comment:
since you're using `StringIO` you shouldn't also write to file.. `StringIO` provides file like object.